src/rest/impl/operations.service.ts

```diff
@@ -100,152 +101,93 @@ export default class OperationsServiceImpl extends OperationsService {
-      const importFiles: SAPImportFile[] = config().sap["import-files"];
-      const datas = await Promise.all(this.parseXmlFiles(importFiles.map((importFile: SAPImportFile) => {
-        return importFile.file;
-      })));
-
-      if (!datas) {
-        this.logger.error("Failed to read SAP import files");
-        return;
-      }
-
-      const sapDatas: SAPData[] = datas.map((data: any, index: number) => {
-        return {
-          data: data.SAP,
-          status: importFiles[index].status
-        };
-      });
-      
-      if (!sapDatas) {
-        this.logger.error("Could not find SAP root entry");
+    switch (type) {
+      case OPERATION_SAP_CONTACT_SYNC:
+        return this.readSapImportBusinessPartners();
+      case OPERATION_SAP_DELIVERY_PLACE_SYNC:
+        return this.readSapImportDeliveryPlaces();
+      case OPERATION_SAP_ITEM_GROUP_SYNC:
+        return this.readSapImportItemGroups();
+      case OPERATION_SAP_CONTRACT_SYNC:
+        return this.readSapImportContracts();
+      default:
         return null;
-      }
-
-      const activeSapData: any = sapDatas.filter((sapData) => {
-        return sapData.status === "APPROVED";
-      }).pop();
-
-      switch (type) {
-        case OPERATION_SAP_CONTACT_SYNC:
-          return this.readSapImportBusinessPartners(activeSapData.data);
-        case OPERATION_SAP_DELIVERY_PLACE_SYNC:
-          return this.readSapImportDeliveryPlaces(activeSapData.data);
-        case OPERATION_SAP_ITEM_GROUP_SYNC:
-          return this.readSapImportItemGroups(activeSapData.data);
-        case OPERATION_SAP_CONTRACT_SYNC:
-          return this.readSapImportContracts(sapDatas);
-      }
-
-    } catch (e) {
-      this.logger.error(`Failed to parse SAP import file ${e}`);
     }
-
-    return null;
   }
 
   /**
-   * Reads business partners from sap and fills related task queue with data from file
-   * 
-   * @param {Object} sap SAP data object 
+   * Reads business partners from sap and fills related task queue with data
    */
-  private async readSapImportBusinessPartners(sap: any) {
-    if (!sap.BusinessPartners) {
-      this.logger.error("Failed to read SAP business parterns");
-      return;
-    }
+  private async readSapImportBusinessPartners() {
+    try {
+      const sapBusinessPartnersService = SapServiceFactory.getBusinessPartnersService();
+      const businessPartners = await sapBusinessPartnersService.listBusinessPartners();
+      const operationReport = await models.createOperationReport("SAP_CONTACT_SYNC");
+
+      businessPartners.forEach((businessPartner: any) => {
+        tasks.enqueueSapContactUpdate(operationReport.id, businessPartner);
+      });
 
-    const businessPartners = sap.BusinessPartners.BusinessPartners;
-    if (!businessPartners) {
-      this.logger.error("Failed to read SAP business parterns list");
+      return operationReport;
+    } catch (e) {
+      this.logger.error(`Failed to read SAP item groups. error: ${e}`);
       return;
```

Comment:
Shoudn't this throw an error?

Reply:
These SAP data operations were not actually try-catched when they are called, there was just check that if these return undefined, the calling function will throw 500, so I figured that I just return undefined then and log the error. Should this be changed?

Reply:
From looking at the original code, it seems that compleltey failed operation will remain in indefinitely "in progress" because items are never marked as "complited". 

Maybe this should be marked as separate issue?